docs/en/query_language/select.md

```diff
@@ -332,42 +334,53 @@ The query can only specify a single ARRAY JOIN clause.
 
 The corresponding conversion can be performed before the WHERE/PREWHERE clause (if its result is needed in this clause), or after completing WHERE/PREWHERE (to reduce the volume of calculations).
 
+<a name="query-language-join"></a>
+
 ### JOIN Clause
 
-The normal JOIN, which is not related to ARRAY JOIN described above.
+Joins the data in the usual [SQL JOIN](https://en.wikipedia.org/wiki/Join_(SQL)) sense.
+
+!!! info "Note"
+    Not related to [ARRAY JOIN](#select-array-join).
+
 
 ``` sql
-[GLOBAL] ANY|ALL INNER|LEFT [OUTER] JOIN (subquery)|table USING columns_list
+SELECT <expr_list>
+FROM <left_subquery>
+[GLOBAL] [ANY|ALL] INNER|LEFT|RIGHT|FULL|CROSS [OUTER] JOIN <right_subquery>
+(ON <expr_list>)|(USING <column_list>) ...
 ```
 
-Performs joins with data from the subquery. At the beginning of query processing, the subquery specified after JOIN is run, and its result is saved in memory. Then it is read from the "left" table specified in the FROM clause, and while it is being read, for each of the read rows from the "left" table, rows are selected from the subquery results table (the "right" table) that meet the condition for matching the values of the columns specified in USING.
+The table names can be specified instead of `<left_subquery>` and `<right_subquery>`. This is equivalent to the `SELECT * FROM table` subquery, except in a special case when the table has the [Join](../operations/table_engines/join.md#table-engine-join) engine – an array prepared for joining.
 
-The table name can be specified instead of a subquery. This is equivalent to the `SELECT * FROM table` subquery, except in a special case when the table has the Join engine – an array prepared for joining.
+**Supported types of `JOIN`**
 
-All columns that are not needed for the JOIN are deleted from the subquery.
+- `INNER JOIN`
+- `LEFT OUTER JOIN`
```

Comment:
It is useful to note, that this optional OUTER is not doing anything, it's just for SQL standard compatibility. Very few people who write queries manually actually waste time on typing it.